framework/src/utils/lib/resources/lambda/create-service-linked-role/index.mjs

```diff
@@ -0,0 +1,22 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import {IAMClient, CreateServiceLinkedRoleCommand} from "@aws-sdk/client-iam"
+
+export const handler = async(event) => {
+    const requestType = event['RequestType']
+
+    if (requestType == "Create") {
+        const client = new IAMClient()
+        const serviceName = event["ResourceProperties"]["serviceName"]
+        
+        try {
+            await client.send(new CreateServiceLinkedRoleCommand({
+                AWSServiceName: serviceName
+            }))
+        } catch (e) {
```

Comment:
can you catch explicitely the error when the role already exists? so any other role will make the deploy fail